src/codeu/chat/server/LocalFile.java

```diff
@@ -0,0 +1,299 @@
+package codeu.chat.server;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileInputStream;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Uuid;
+import codeu.chat.util.Serializer;
+import codeu.chat.util.Serializers;
+
+public class LocalFile
+{
+    private final static Logger.Log LOG = Logger.newLog(LocalFile.class);
+
+    public static final String MESSAGE_FILE_NAME = "/msgDat.sav";
+    public static final String USER_FILE_NAME = "/usrDat.sav";
+    public static final String CONVERSATION_FILE_NAME = "/cvrsDat.sav";
+
+    //Instance varibles for saving the current data of server.
+    private final LinkedHashSet<User> users;
+    private final LinkedHashSet<ConversationHeader> conversationHeaders;
+    private final LinkedHashSet<Message> messages;
+
+    private final File file;
+
+    private boolean hasUserModified = false;//It indicates if there is a new data should be handled.
+    private boolean hasMessageModified = false;
+    private boolean hasConversationModified = false;
+
+    //false: The instance is not initialized. All new data will not mark the status as modified.
+    //true: All new data been added to this instance should mark the status as modified.
+    private boolean isInitialized = false;
```

Comment:
You should delete `isInitialized` here as well. This class should always be initialized if its constructor completes successfully.